### PR TITLE
Add PPP support

### PR DESCRIPTION
**Pull request by jonasj76**
Add support for kernel PPP interfaces.

### CODE REVIEW

File: lib/route/link/ppp.c [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ * @route_doc{link_ppp, PPP Documentation}
+ * @{
+ */
+
+#include <netlink-private/netlink.h>
+#include <netlink/netlink.h>
+#include <netlink-private/route/link/api.h>
+
+/** @cond SKIP */
+#define PPP_HAS_FD		(1<<0)
```
Comment (thom311) [inline lib/route/link/ppp.c]: could we name this PPP_ATTR_FD for consistency with other types?

File: lib/route/link/ppp.c [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+struct ppp_info
+{
+	uint32_t		pi_fd;
+	uint32_t		pi_mask;
```
Comment (thom311) [inline lib/route/link/ppp.c]: most other implementations name this field ce_mask. Could we stick to that?

File: lib/route/link/ppp.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	return 0;
+}
+
+/**
+ * Get PPP file descriptor
+ * @arg link		Link object
+ *
+ * @return PPP file descriptor, 0 if not set or a negative error code.
+ */
+uint32_t rtnl_link_ppp_get_mode(struct rtnl_link *link)
```
Comment (thom311) [inline lib/route/link/ppp.c]: should be named rtnl_link_ppp_get_fd() for consistency.

Also, I know libnl3's API is quite inconsistent, but I would prefer as signature
    int rtnl_link_ppp_get_fd (struct rtnl_link *link, uint32 *out_fd);
to return errors separate from the value.

**Comment by jonasj76** [comment]
Updated code after review.

Also:
- Renaming local `struct ppp_info*` variables from `pi` -> `info`.
- Fix type of file descriptor: `uint32_t` -> `int32_t`.

**Comment by thom311** [comment]
merged as https://github.com/thom311/libnl/commit/306695fb07a39a0e76b361ce62f5f38c28cf4f79 with one minor fix at the end.

-- also, during rebase, there were some conflicts that I resolved, so the patches are not 100 % original...


thanks!!

**Pull request closed without merging**